render/render.go

```diff
@@ -42,9 +63,32 @@ func Render(fileName string, inputData []byte) error {
 	}
 	defer output.Close()
 
-	if err = tmpl.Execute(output, results); err != nil {
+	if err = tmpl.ExecuteTemplate(output, "html.tpl", results); err != nil {
 		return xerrors.Errorf("error executing template: %v\n", err)
 	}
 
 	return nil
 }
+
+func collectFiles(templateFS fs.FS) ([]string, error) {
+	var files []string
+	err := fs.WalkDir(templateFS, ".", func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return xerrors.Errorf("error listing files in %s: %w", path, err)
+		}
+
+		for _, ext := range extensions {
+			if !d.IsDir() && strings.HasSuffix(path, ext) {
```

Comment:
If it is a directory, you can do an early return immediately after error checking